STUDIO/org.openl.rules.repository.azure/src/org/openl/rules/repository/azure/AzureBlobRepository.java

```diff
@@ -0,0 +1,760 @@
+            throw e;
+        } catch (Exception e) {
+            throw new IOException(e);
+        }
+
+        return fileData;
+    }
+
+    @Override
+    public String getId() {
+        return id;
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public List<FileData> list(String path) throws IOException {
+        try {
+            BlobClient client = findCommitBlob(path, null);
+            if (client != null) {
+                return getFilesForCommit(getCommit(client));
+            } else {
+                return Collections.emptyList();
+            }
+        } catch (Exception e) {
+            throw new IOException(e);
+        }
+    }
+
+    @Override
+    public FileData check(String name) throws IOException {
+        try {
+            final BlobClient client = blobContainerClient.getBlobClient(CONTENT_PREFIX + name);
+            if (client.exists()) {
+                // File
+                return createFileDataForFile(name, null, client);
+            } else {
+                // Project folder or sub-folder.
+                return createFileDataForFolder(name, null);
+            }
+        } catch (Exception e) {
+            throw new IOException(e);
+        }
+    }
+
+    @Override
+    public FileItem read(String name) throws IOException {
+        try {
+            final BlobClient client = blobContainerClient.getBlobClient(CONTENT_PREFIX + name);
+            if (client.exists()) {
+                FileData fileData = createFileDataForFile(name, null, client);
+                return  new FileItem(fileData, client.openInputStream());
+            }
+
+            return null;
+        } catch (Exception e) {
+            throw new IOException(e);
+        }
+    }
+
+    @Override
+    public FileData save(FileData data, InputStream stream) throws IOException {
+        throw new UnsupportedOperationException(UNSUPPORTED_IN_FOLDER_REPOSITORY);
+    }
+
+    @Override
+    public List<FileData> save(List<FileItem> fileItems) throws IOException {
+        throw new UnsupportedOperationException(UNSUPPORTED_IN_FOLDER_REPOSITORY);
+    }
+
+    @Override
+    public boolean delete(FileData data) throws IOException {
+        try {
+            String path = data.getName();
+
+            AzureCommit commit = new AzureCommit();
+            commit.setAuthor(data.getAuthor().getUsername());
+            commit.setComment(data.getComment());
+            commit.setDeleted(true);
+            commit.setModifiedAt(new Date());
+
+            saveCommit(commit, path);
+
+            onModified();
+
+            return true;
+        } catch (IOException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new IOException(e);
+        }
+    }
+
+    @Override
+    public boolean delete(List<FileData> data) throws IOException {
+        throw new UnsupportedOperationException(UNSUPPORTED_IN_FOLDER_REPOSITORY);
+    }

```

Comment:
Why aren't batch operations supported?

Reply:
This method is invoked only for Repository that doesn't support folder structure. Azure Blob Repository supports is designed to work as a folder repository for that case delete/deleteHistory for base folder must be invoked, so this method isn't needed in our case.